roboteam_ai/src/interface_api/InterfaceSubscriber.cpp

```diff
@@ -47,21 +50,27 @@ void InterfaceSubscriber::onMessage(const proto::MsgFromInterface&& message) {
                 Pause::resume();
             }
         } break;
+        
         case proto::MsgFromInterface::kSetBallPos: {
-            const auto& ballPos = message.set_ball_pos();
-            proto::SimulationConfiguration configuration;                    // Create packet
-            configuration.mutable_ball_location()->set_x(ballPos.x());  // Set x
-            configuration.mutable_ball_location()->set_y(ballPos.y());  // Set y
+            RTT_WARNING("TODO: Remove kSetBallPos from MsgFromInterface")
+        } break;
 
-            bool sentConfig = io::io.sendSimulationConfiguration(configuration);  // Send packet
-            if (!sentConfig) {
-                RTT_WARNING("Failed to send Simulation Configuration command. Is this the primary AI?")
-            }
+        case proto::MsgFromInterface::kSimulatorCommand: {
+            const SimulatorCommand& command = message.simulator_command();
+            sendPacketToSimulator(command);
         } break;
+
         case proto::MsgFromInterface::KIND_NOT_SET:
             RTT_ERROR("Received message with no kind set");
             break;
     }
 }
 
+void InterfaceSubscriber::sendPacketToSimulator(const SimulatorCommand& packet) {
+    QByteArray datagram;
+    datagram.resize(packet.ByteSizeLong());
+    packet.SerializeToArray(datagram.data(), datagram.size());
+    auto bytesSent = simulator_socket.writeDatagram(datagram, QHostAddress::LocalHost, 10300);
```

Comment:
I don't like that the UI subscriber has dependency on QT. What about moving this to IOManager and just invoking it in the switch here?

Reply:
Done